code-challenges.test.js

```diff
@@ -15,35 +15,136 @@
+        expect(["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]).toEqual(expect.arrayContaining(expected2))
+    });
+    it('does not match if received has "chartreuse" in the array', () => {
+        expect(["chartreuse"]).not.toEqual(expect.arrayContaining(expected2))
+    });
+});
 
 // b) Create the function that makes the test pass.
-
+//PsuedoCode
+//A function using the .shift() method, which would remove the value at the zeroth index ("purple" and "chartreuse") followed up by .sort() to put the values in the array in a random. The syllabus states that there are advanced sorting methods. After doing some research, I found using a randomIndex variable utilizing Math.floor(Math.random() * the current array) after .shift() might possibly work. Since I defined a result in the test, I hope "shuffling" the remain array values will still make the test pass. We've worked with randomizing numbers/ values in the past in class so using Math.floor(Math.random()) in some capacity  will probably be needed. 
+//Parameters:
+    //Input: //const colors1 = ["purple", "blue", "green", "yellow", "pink"], //const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+//Output: Expected output = ["pink", "blue", "yellow", "green"]; Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+
+//I tried using the .shift() method but quickly realized that the original array would be affected so I was only getting "purple" as an output. Even when I .sort() the original array, I came back with "blue" as the output. So researching shuffling methods, random methods, I came across the Fisher Yates Method. I tested out the example function that W3 has. The issue I am having is trying to figure out how to get rid of the [0] color. I used .slice(0,1) in the console.log portion in hopes of getting back every color but purple together but I got back every color but purple, seperately. After using the .shift() again but declaring a new variable, newArr = colors1.shift() and put it above the Fisher Yates method, I was able to get purple to not be part of the output. 
+    //I had to go back and rework my test once I knew my function worked. I didn't comprehend what the example provided was stating. But after refactoring the values in the arrays, the test passed!
+
+const colors1 = ["purple", "blue", "green", "yellow", "pink"];
+    newArr = colors1.shift()
+    
+    for (let i = colors1.length -1; i > 0; i--) {
+      let j = Math.floor(Math.random() * i)
+      let k = colors1[i]
+      colors1[i] = colors1[j]
+      colors1[j] = k
+    }
+    console.log(colors1)
+
+const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"];
+    newArr = colors2.shift()
+
+for (let i = colors2.length -1; i > 0; i--) {
+  let j = Math.floor(Math.random() * i)
+  let k = colors2[i]
+  colors2[i] = colors2[j]
+  colors2[j] = k
+}
```

Comment:
The logic on this problem looks good, but this is currently not set up as a function.  A function will allow you to use the same block of code over and over for different test variables and not need to repeat it